mailpoet/tests/performance/tests/subscribers-filtering.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-shadow */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable import/no-default-export */
+/**
+ * External dependencies
+ */
+import { sleep, check, group } from 'k6';
+import { chromium } from 'k6/x/browser';
+import { randomIntBetween } from 'https://jslib.k6.io/k6-utils/1.1.0/index.js';
+
+/**
+ * Internal dependencies
+ */
+import {
+  baseURL,
+  thinkTimeMin,
+  thinkTimeMax,
+  headlessSet,
+  adminEmail,
+} from '../config.js';
+import { authenticate } from '../utils/helpers.js';
+/* global Promise */
+
+export function subscribersFiltering() {
+  const browser = chromium.launch({ headless: headlessSet });
+  const page = browser.newPage();
+
+  group('Subscribers - Filter subscribers', function subscribersFiltering() {
+    page
+      .goto(`${baseURL}/wp-admin/admin.php?page=mailpoet-subscribers`, {
+        waitUntil: 'networkidle',
+      })
+
+      .then(() => {
+        authenticate(page);
+      })
+
+      .then(() => {
+        return Promise.all([
+          page.waitForNavigation({ waitUntil: 'networkidle' }),
+        ]);
+      })
+
+      .then(() => {
+        page.locator('[data-automation-id="filters_subscribed"]').click();
+        page.waitForSelector('[data-automation-id="filters_subscribed"]');
+        check(page, {
+          'subscribers filter is visible': page
+            .locator('[data-automation-id="listing_filter_segment"]')
+            .isVisible(),
+        });
+      })
+
+      .then(() => {
+        page
+          .locator('[data-automation-id="listing_filter_segment"]')
+          .selectOption('3');
+        sleep(2);
```

Comment:
I'm not sure I understand this sleep. Is it to ensure the ajax request has started? Maybe instead you could wait for the Loading subscribers text?
<img width="206" alt="image" src="https://user-images.githubusercontent.com/8656640/220173114-761b1ab9-7133-47be-9066-77a4d108a923.png">

Reply:
I can try that, yes. The reason was to trigger this loading state, and adding sleeps is widely accepted in performance testing, it is even recommended to use it.

Reply:
My concern was that the sleep of 2 seconds might not always be enough time in some rare cases to trigger the loading state, which could lead to unexpected results. Probably not likely, or maybe not even possible. I just always prefer to wait for something specific if it's feasible instead of relying on wait times. Not only can it be more reliable, but it also makes the test code more self documenting -- someone reading it would immediately understand the state you're waiting for.

It's interesting to hear that sleeps are encouraged in performance testing, though, and I'm interested to hear how it might be useful in this case other than waiting to trigger the loading state.

Reply:
Yes I agree with you, I don't want to use sleeps and will try to avoid it as much as possible. With much larger sleeps of 5 to 10 seconds, that's how they slow down the tests but we have mission to have this test suite running fast. I can ask this question about using sleeps often in tests in some of the upcoming meetings with k6 happy hours.